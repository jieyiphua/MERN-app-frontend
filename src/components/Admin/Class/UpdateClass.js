import React from "react";
import { Redirect, withRouter, Link } from "react-router-dom";
import axios from "axios";

class UpdateClass extends React.Component {
  state = {
    data: null,
    selectedContent: null,
    name: this.props.selectedContent.name || "",
    description: this.props.selectedContent.description || "",
    category: this.props.selectedContent.category || "",
    image: this.props.selectedContent.image || "",
    id: this.props.selectedContent._id,
    errors: []
  };

  onInputChange = event => {
    this.setState({ [event.target.name]: event.target.value });
    console.log(this.state);
  };

  onFormSubmit = async event => {
    event.preventDefault();
    try {
      console.log(this.state);
      const response = await axios.put(
        `https://deployment-mern-backend-tessivanjayz.gardtess.now.sh/class/${
          this.state.id
        }`,
        {
          name: this.state.name,
          description: this.state.description,
          category: this.state.category,
          image: this.state.image
        }
      );
      console.log(response);
    } catch (error) {
      this.setState({ errors: error.response });
    }
    this.props.history.push("/classes");
  };

  render() {
    const { authentication } = this.props;

    if (!authentication) {
      return <Redirect to="/admin/login" />;
    }

    return (
      <div className="main-container" id="new-class-main-container">
        <div className="new-class-header">
          <h1>Update Class</h1>
        </div>
        <div className="new-class-form-container">
          <form
            className="new-class-form"
            onSubmit={this.onFormSubmit}
            style={{ display: "flex", flexDirection: "column" }}
          >
            <label id="first-label" htmlFor="name">
              Name:
            </label>
            <input
              type="text"
              name="name"
              value={this.state.name}
              placeholder="Name"
              onChange={this.onInputChange}
            />
            <label htmlFor="description"> Description:</label>
            <textarea
              rows="10"
              cols="50"
              name="description"
              placeholder="Enter Description"
              value={this.state.description}
              onChange={this.onInputChange}
            />
            <label htmlFor="faculty">Category: </label>
            <input
              type="text"
              name="category"
              placeholder="Category"
              value={this.state.category}
              onChange={this.onInputChange}
            />
            <label htmlFor="faculty">Image: </label>
            <input
              type="text"
              name="image"
              placeholder="Image URL"
              value={this.state.image}
              onChange={this.onInputChange}
            />
            <div className="classes-buttons">
              <button onClick={this.onFormSubmit}>Submit</button>
              <Link to="/classes">
                <button>Back</button>
              </Link>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

export default withRouter(UpdateClass);
