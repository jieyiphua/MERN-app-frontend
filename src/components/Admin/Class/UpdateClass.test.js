import UpdateClass from "./UpdateClass";
import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon from "sinon";

describe("<UpdateClass />", () => {
  it("renders component", () => {
    const wrapper = shallow(<UpdateClass />);
    expect(wrapper).toMatchSnapshot();
  });
});
