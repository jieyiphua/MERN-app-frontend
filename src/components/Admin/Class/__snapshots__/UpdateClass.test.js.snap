// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateClass /> renders component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
